pkg/raft/fsm/fsm.go

```diff
@@ -0,0 +1,102 @@
+package fsm
+
+import (
+	"bufio"
+	"encoding/base64"
+	"io"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/hashicorp/raft"
+	"github.com/rs/zerolog/log"
+)
+
+type CommunityStateMachine struct {
+	Log       []string `json:"log"`
+	StateJSON []byte
+}
+
+func NewCommunityStateMachine() *CommunityStateMachine {
+	return &CommunityStateMachine{
+		Log:       make([]string, 0),
+		StateJSON: []byte("{}"),
+	}
+}
+
+// Apply log is invoked once a log entry is committed.
+// It returns a value which will be made available in the
+// ApplyFuture returned by Raft.Apply method if that
```

Comment:
what is ApplyFuture?

Reply:
Since the value in the log is not immediately committed by the Raft cluster, Raft needs a way to describe the future expected state without actually guaranteeing that it has been confirmed yet.